lib/auth/auth.go

```diff
@@ -3347,6 +3352,58 @@ func (a *Server) DeleteNetworkRestrictions(ctx context.Context) error {
 	return a.Services.Restrictions.DeleteNetworkRestrictions(ctx)
 }
 
+// GetPluginData loads all plugin data matching the supplied filter.
+func (a *Server) GetPluginData(ctx context.Context, filter types.PluginDataFilter) ([]types.PluginData, error) {
+	return a.Services.PluginData.GetPluginData(ctx, filter)
+}
+
+// GetPluginData loads all plugin data matching the supplied filter.
+func (a *Server) UpdatePluginData(ctx context.Context, params types.PluginDataUpdateParams) error {
+	// Load parent resource expiration value. Nil means that a parent resource does not exist.
+	// Non-existent parent resource is fine on update. There are the cases when we need to access
+	// PluginData of a just deleted resource. We need to prevent creation of PluginData entries
+	// for a missing resources.
+	//
+	// Expires itself is needed to prevent orphaned plugin data, we automatically
+	// configure new instances to expire shortly after the resource to which they are associated.
```

Comment:
nit: I don't think this comment is necessary, this is explained in detail on the backend.
```suggestion
```